.github/workflows/perf-testing-setup.yml

```diff
@@ -47,13 +38,63 @@ jobs:
         python setup.py extension
     - name: Build and Run the Docker image
       run: |
+        echo "Building Docker image with Python version ${{ env.PYTHON_VERSION }}"
         docker build --build-arg PYTHON_VERSION=${{ env.PYTHON_VERSION }} --file .ci/perf_tests/dockerfiles/perf_tests.Dockerfile --tag perfimage:latest .
-        docker run -d --shm-size="2g" --env FUNCTIONS_WORKER_RUNTIME_VERSION=${{ env.PYTHON_VERSION }} -p ${PORT}:80 -v $GITHUB_WORKSPACE/azure_functions_worker:/azure-functions-host/workers/python/${{ env.PYTHON_VERSION }}/LINUX/X64/azure_functions_worker perfimage:latest
+    
+        echo "Running Docker container..."
+        container_id=$(docker run -d --privileged --shm-size="2g" --env FUNCTIONS_WORKER_RUNTIME_VERSION=${{ env.PYTHON_VERSION }} -p ${PORT}:80 -v $GITHUB_WORKSPACE/azure_functions_worker:/azure-functions-host/workers/python/${{ env.PYTHON_VERSION }}/LINUX/X64/azure_functions_worker perfimage:latest)
         sleep 10  # host needs some time to start.
+        echo "Container ID is $container_id"
+        
+        echo "Fetching Docker container logs..."
+        docker logs $container_id
+        worker_pid=$(docker exec $container_id sh -c "ps aux | grep '[p]ython'" | awk '{print $2}')
```

Comment:
Is this intentional? [p]ython

Reply:
yes [p] is evaulated in grep using regex, makes sure grep is only looking for python process rather than itself("ps aux | grep 'python')